php/Terminus/Collections/UserOrganizationMemberships.php

```diff
@@ -30,4 +30,29 @@ public function __construct($options = [])
         $this->user = $options['user'];
         $this->url  = "users/{$this->user->id}/memberships/organizations";
     }
+
+    /**
+     * Retrieves a member by either by the membership ID, the member organization ID, or the member organization name
+     *
+     * @param string $id Either a membership or member-org UUID or a member-org name
+     * @return UserOrganizationMembership
+     * @throws TerminusException
+     */
+    public function get($id)
+    {
+        $models = $this->getMembers();
+        if (isset($models[$id])) {
+            return $models[$id];
+        }
+        foreach ($models as $model) {
+            $organization = $model->organization;
+            if (in_array($id, [$organization->get('profile')->name, $organization->id,])) {
+                return $model;
+            }
+        }
+        throw new TerminusException(
+            'An organizational member identified by "{id}" could not be found.',
+            compact('id')
+        );
```

Comment:
We have a `TerminusNotFound` exception for exactly this:

https://github.com/pantheon-systems/terminus/blob/master/php/Terminus/Exceptions/TerminusNotFoundException.php